### ISSUE TITLE
关于开源代码的一些问题

### ISSUE THREAD

**Issue by Alloman** [Jinxi-Yang-WHU/Fast-GCR#2]
```python
    temp = torch.sum(S, dim=0)
    # print(temp.min(), temp.max(), temp.shape)
    D_row = torch.sqrt(1. / torch.sum(S, dim=1))
    D_col = torch.sqrt(1. / torch.sum(S, dim=0))
    L = torch.outer(D_row, D_col) * S
    global_X = L @ X  # 使用 @ 符号进行矩阵乘法

    X = cross_X 
    if _cfg.GCR.MODE != 'no-norm':
        X = X / torch.linalg.norm(X, ord=2, dim=1, keepdims=True)  # 使用 torch.linalg.norm 进行归一化
    if _cfg.COMMON.VERBOSE:
        print(f'round time {time.time() - start_time} s')
    return X
```
代码中的这一部分出现了一个cross_X,这个变量在其他地方都没有出现。并且计算后的global_x也没有被使用， 不知道这是否是一个错误

**Comment by Jinxi-Yang-WHU** [comment]
这确实是一个错误，非常感谢您能指出这个错误，正确的代码应该是X = global_X。


-----原始邮件-----
发件人:"WANG ZHAOFA" ***@***.***>
发送时间:2025-07-15 13:56:37 (星期二)
收件人: Jinxi-Yang-WHU/Fast-GCR ***@***.***>
抄送: Subscribed ***@***.***>
主题: [Jinxi-Yang-WHU/Fast-GCR] 关于开源代码的一些问题 (Issue #2)


Alloman created an issue (Jinxi-Yang-WHU/Fast-GCR#2)

`
temp = torch.sum(S, dim=0)
# print(temp.min(), temp.max(), temp.shape)
D_row = torch.sqrt(1. / torch.sum(S, dim=1))
D_col = torch.sqrt(1. / torch.sum(S, dim=0))
L = torch.outer(D_row, D_col) * S
global_X = L @ X # 使用 @ 符号进行矩阵乘法

X = cross_X 
if _cfg.GCR.MODE != 'no-norm':
    X = X / torch.linalg.norm(X, ord=2, dim=1, keepdims=True)  # 使用 torch.linalg.norm 进行归一化
if _cfg.COMMON.VERBOSE:
    print(f'round time {time.time() - start_time} s')
return X


`
代码中的这一部分出现了一个cross_X,这个变量在其他地方都没有出现。并且计算后的global_x也没有被使用， 不知道这是否是一个错误
